code/modules/mob/living/carbon/xenomorph/castes/hivemind/abilities_hivemind.dm

```diff
@@ -5,3 +5,16 @@
 
 /datum/action/xeno_action/return_to_core/action_activate()
 	SEND_SIGNAL(owner, COMSIG_XENOMORPH_CORE_RETURN)
+
+/datum/action/xeno_action/activable/secrete_resin/ranged/slow
+	cooldown_timer = 5 SECONDS
+	base_wait = 2.5 SECONDS
+	max_range = 5
+
+/datum/action/xeno_action/change_form
+	name = "Change form"
+	action_icon_state = "manifest"
+	mechanics_text = "Change from your incorporal form to your physical on and vice-versa."
+
+/datum/action/xeno_action/change_form/action_activate()
+	SEND_SIGNAL(owner, COMSIG_XENOMORPH_HIVEMIND_CHANGE_FORM)
```

Comment:
it's either a signal or a cast of owner into hivemind. I don't like the former, because the design philosophy of abilities is to be portable. If I have to caste, this become snowflake ability.

This is why the other hivemind ability, return to core, uses a signal (and it was made by spooky)